tests/integration/gcp.py

```diff
@@ -440,21 +436,20 @@ def _get_image(self, project, image_name):
         self.logger.debug(f"Looking for {image_name=} in {project=}")
         try:
             response = (
-                self._compute.images().get(project=project, image=image_name,).execute()
+                self._compute_images.get(project=project, image=image_name,)
             )
-            image = response.get("selfLink")
+            image = response.self_link
             return image
         except:
             self.logger.debug(f"Failed to find image in {project=}")
             return None
 
     def _wait_until_reachable(self, hostname):
-        self.logger.info(f"Waiting for {hostname} to respond to ping ...")
+        self.logger.info(f"Waiting for {hostname} to respond...")
         while True:
-            response = os.system("ping -c 1 " + hostname)
+            response = os.system("timeout 1 bash -c \"</dev/tcp/" + hostname + "/22\"")
             if response == 0:
-                self.logger.info(f"Instance {hostname} is reachable, waiting another 20 seconds...")
-                time.sleep(20)
+                self.logger.info(f"Instance {hostname} is reachable...")
                 return
             time.sleep(1)
```

Comment:
Although we old implementation didn't had this as well, we should probably add some sort of retry counter, because otherwise this function could hang forever, if the SSH port never comes up. We should then use a reasonable default for this retry counter.